service/auth/Dockerfile

```diff
@@ -0,0 +1,5 @@
+FROM scratch
+LABEL image="ghcr.io/labring/sealos/service/auth"
+COPY main ./main
+COPY auth.yaml auth.yaml
+CMD ["./main", "-config", "./auth.yaml"]
```

Comment:
Maybe we can use multi-stage dockerfile build here?
Golang build is the first stage, and here is the second and final stage

Reply:
SelfMake using docker file? or just run make file? I'm not sure which is better for global mantance.
Butirst time we can use docker file to build it, and combining outputs to cluster yaml?
@SignorMercurio Any suggestion?

Reply:
We can still make use of Makefile in the first stage like `CMD ["make", "build", "auth"]
This will simplify the release process into one step (directly build the image and push it to dockerhub). Otherwise, we have to prepare a golang environment first to build the binary file and then we can build the image.

I assume that the release of binary and docker images are two individual CI processes. If we release them together, I agree with current dockerfile.

Reply:
Multi-stage building is a cool feature, please feel free to contribute. But at present, binary releasing and docker image publishing are within a single CI process (see .goreleaser.yml), and I'm not sure whether we should separate them because doing so needs to tamper with GoReleaser configs.

Reply:
@zzjin I'm not familiar with this module, but I believe `service/auth` here produces a binary. According to the existing directory structure, should we move these into `cmd/`? From the `Dockerfile` I guess that the binary is named `main`, should we change it? Finally, I suggest moving any `Dockerfile` to `docker/` so that we can possibly `make image.build` and `make image.push` in the future.

Reply:
At this first stage, packages(currently only `auth`, but will add more soon) under `service` first build one binary, and combine this binary and config file(s) with many other requirements into a full cluster image.

So we should push the cluster image to hub but not the binary docker file itself.
In the other words, we should not push(or publish) binary's Dockerfile.

Reply:
You can directly make a cluster image, start a registry locally, and push the mirror to the mirror repository. The final product is cluster mirroring.